### PR TITLE
[feat] add an option to specify the file name prefix for all metric f…

### PR DESCRIPTION
**Pull request by nh13**
…iles

### CODE REVIEW

File: src/lib/metrics.rs [hunk 1]
```diff
@@ -174,8 +174,15 @@ impl UnmatchedCounter {
     }
 
     /// Write the top `n` unmatched barcodes to a `most_frequent_unmatched.tsv` file in the specified directory.
-    pub fn to_file<P: AsRef<Path>>(self, output_dir: P, n: usize) -> Result<()> {
-        let output_path = output_dir.as_ref().join("most_frequent_unmatched.tsv");
+    pub fn to_file<P: AsRef<Path>>(
+        self,
+        output_dir: P,
+        n: usize,
+        prefix: Option<String>,
```
Comment (nh13) [inline src/lib/metrics.rs:181]: there's got to be a better way to pass this around than `Option<String>` so I can concatenate below.

File: src/lib/metrics.rs [hunk 2]
```diff
@@ -174,8 +174,16 @@ impl UnmatchedCounter {
...
+    ) -> Result<()> {
+        let filename =
+            [prefix.unwrap_or_else(|| "".to_string()), "most_frequent_unmatched.tsv".to_string()]
+                .concat();
```
Comment (tfenne) [inline src/lib/metrics.rs:185]:
```suggestion
        let filename = format!("{}{}", prefix.unwrap_or_default(), "most_frequent_unmatched.tsv");

```

File: src/lib/run.rs [hunk 3]
```diff
@@ -304,9 +304,13 @@ pub fn run(opts: Opts) -> Result<()> {
     pool.stop_pool()?;
 
     info!("Writing stats");
-    metrics.write_metrics_files(&samples, &opts.output_dir)?;
+    metrics.write_metrics_files(&samples, &opts.output_dir, opts.metric_prefix.clone())?;
```
Comment (tfenne) [inline src/lib/run.rs:307]: Here's where I would be tempted to do:

```rust
let metrics_prefix = &opts.metrics_prefix.map(|s| s.as_str()).unwrap_or("");
```
and then use that everywhere.

**Review by tfenne — APPROVED** [review]
Two conflicting suggestions - take your pick!

**Review by omicsorama — APPROVED** [review]

**Pull request merged**